plugin-viewport/src/main/kotlin/com/mapbox/maps/plugin/viewport/experimental/transition/MapboxViewportTransitionFactory.kt

```diff
@@ -0,0 +1,278 @@
+
+  fun transitionFromLowZoomToHighZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
+    val currentMapCameraState = cameraManager.cameraState
+
+    var centerDuration = 0L
+    cameraOptions.center?.let { center ->
+      val screenDistanceFromMapCenterToLocation =
+        screenDistanceFromMapCenterToTarget(
+          cameraManager = cameraManager,
+          currentCenter = currentMapCameraState.center,
+          targetCenter = center
+        )
+
+      val centerAnimationRate = 500L
+      centerDuration =
+        ((screenDistanceFromMapCenterToLocation / centerAnimationRate) * 1000L).toLong()
+          .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION_MS)
+      animators += createCenterAnimator(
+        center = center,
+        duration = centerDuration
+      )
+    }
+
+    var zoomDelay = 0L
+    var zoomDuration = 0L
+    cameraOptions.zoom?.let { zoom ->
+      val currentMapCameraZoom = currentMapCameraState.zoom
+      val zoomDelta = abs(zoom - currentMapCameraZoom)
+      val zoomAnimationRate = 2.2
+      zoomDelay = centerDuration / 2L
+      zoomDuration = ((zoomDelta / zoomAnimationRate) * 1000L).toLong()
+        .coerceAtMost(MAXIMUM_LOW_TO_HIGH_DURATION_MS)
+
+      animators += createZoomAnimator(
+        zoom = zoom,
+        startDelay = zoomDelay,
+        duration = zoomDuration
+      )
+    }
+
+    cameraOptions.bearing?.let { bearing ->
+      val bearingDuration = 1800L
+      val bearingDelay = (zoomDelay + zoomDuration - bearingDuration).coerceAtLeast(0L)
+      animators += createBearingAnimator(
+        rotation = normalizeBearing(
+          currentMapCameraState.bearing,
+          bearing
+        ),
+        startDelay = bearingDelay,
+        duration = bearingDuration
+      )
+    }
+
+    val pitchAndPaddingDuration = 1200L
+    val pitchAndPaddingDelay = (zoomDelay + zoomDuration - pitchAndPaddingDuration + 100L)
+      .coerceAtLeast(0L)
+    cameraOptions.pitch?.let { pitch ->
+      animators += createPitchAnimator(
+        pitch = pitch,
+        startDelay = pitchAndPaddingDelay,
+        duration = pitchAndPaddingDuration
+      )
+    }
+
+    cameraOptions.padding?.let { padding ->
+      animators += createPaddingAnimator(
+        padding = padding,
+        startDelay = pitchAndPaddingDelay,
+        duration = pitchAndPaddingDuration
+      )
+    }
+
+    return createAnimatorSet(animators).constrainDurationTo(transitionMaxDurationMs)
+  }
+
+  fun transitionFromHighZoomToLowZoom(
+    cameraOptions: CameraOptions,
+    transitionMaxDurationMs: Long
+  ): AnimatorSet {
+    val animators = mutableListOf<Animator>()
+    val currentMapCameraState = cameraManager.cameraState
+
+    cameraOptions.center?.let { center ->
+      animators += createCenterAnimator(
+        center = center,
+        startDelay = 800L,
```

Comment:
how about moving this values to constants? 
And I, as a user, how can I change this constants?

Reply:
This is the predefined `DefaultViewportTransition` and is an implementation detail. We don't expose config options to customise these yet, as the property list would be huge if we add every parameter. We could add more configs to our public interface later on incrementally if there's a lot of user request, for now let's keep it simple as an out-of-box solution, and user could implement their own ViewportTransition if they want to customise the transition animation heavily.